### DISCUSSION TITLE
Building AppVeyor builds for extensions

### DISCUSSION THREAD

**Discussion by GeertvanHorrik** [cake-build/cake#4325] [Q&A] [answered]
I am using Cake.Recipe and updating a (long overdue) project (Cake.GitHub) to make sure it can be released as 1.0.0. The local builds work fine:

```
.\build.ps1 --target=CI
```

But the build on AppVeyor fails with this error:

(see https://ci.appveyor.com/project/cakecontrib/cake-github/builds/50389211)

```
C:\projects\cake-github\src\Cake.GitHub\obj\Release\net6.0\Cake.GitHub.AssemblyInfo.cs(17,12): error CS0579: Duplicate 'System.Reflection.AssemblyFileVersionAttribute' attribute [C:\projects\cake-github\src\Cake.GitHub\Cake.GitHub.csproj::TargetFramework=net6.0]
C:\projects\cake-github\src\Cake.GitHub\obj\Release\net6.0\Cake.GitHub.AssemblyInfo.cs(18,12): error CS0579: Duplicate 'System.Reflection.AssemblyInformationalVersionAttribute' attribute [C:\projects\cake-github\src\Cake.GitHub\Cake.GitHub.csproj::TargetFramework=net6.0]
C:\projects\cake-github\src\Cake.GitHub\obj\Release\net6.0\Cake.GitHub.AssemblyInfo.cs(21,12): error CS0579: Duplicate 'System.Reflection.AssemblyVersionAttribute' attribute [C:\projects\cake-github\src\Cake.GitHub\Cake.GitHub.csproj::TargetFramework=net6.0]
```

I suspect that both the recipe and GitVersion are updating the AssemblyInfo, but have no idea how to fix this using the recipe. Hopefully someone experienced with the recipe (@gep13?) can help me out here.

**Comment by pascalberger** [comment, marked as answer]
You need to rename the `AssemblyInfo.cs` file in your project to a different file name. See here for an example: https://github.com/cake-contrib/Cake.Issues/blob/develop/src/Cake.Issues.DocFx/Properties/ProjectInfo.cs

**Reply by GeertvanHorrik** [reply]
It works, thank you! Not sure why it behaves differently on AV than on my local machine, but thanks a lot!

**Reply by GeertvanHorrik** [reply]
@pascalberger since you seem to know how this works, do you know why it doesn't release / publish a prerelease nuget package? I would like to release a beta before doing an actual stable release.

**Reply by gep13** [reply]
Do you have a link to a build on AppVeyor where it isn't doing what you expect?

**Reply by GeertvanHorrik** [reply]
See https://ci.appveyor.com/project/cakecontrib/cake-github/builds/50389445

I tried manually running a deploy, but there doesn't seem to be a target.

**Reply by gep13** [reply]
The issue is here: https://ci.appveyor.com/project/cakecontrib/cake-github/builds/50389445#L489

There are no nupkgs being generated, so as a result, there is nothing to publish to the configured pre-release feed.

In this [commit](https://github.com/cake-contrib/Cake.GitHub/commit/5ef0fe09303b60758e6630df0ff5111e85783dfd), you removed `shouldRunDotNetCorePack: true,`.  Adding that back in should make the nupkg be created again.

**Reply by GeertvanHorrik** [reply]
Thank you @gep13 , that was it! Thank you both very much for your time and wisdom

**Reply by gep13** [reply]
Happy to help! 😄

**Reply by GeertvanHorrik** [reply]
I just "released"1.0.0 (see https://ci.appveyor.com/project/cakecontrib/cake-github/builds/50392883). It's not (yet) on NuGet, will it automatically appear or do I need to execute some specific action? Then the component should be in good shape and can be maintained and released from here.

**Reply by gep13** [reply]
When you say you "released", can you confirm what you mean here? A tag hasn't been applied, at least, it wasn't a tag that triggered the AppVeyor build.  You can see that here:

https://ci.appveyor.com/project/cakecontrib/cake-github/builds/50392883#L833

**Reply by gep13** [reply]
Ah... how did you apply the tag? Was it done through the GitHub UI, or did you create the tag locally and push to GitHub?

The latter doesn't trigger the webhook to AppVeyor, so the CI build doesn't "see" the tag.

**Reply by gep13** [reply]
The process I typically follow is documented here: https://cake-contrib.github.io/Cake.Recipe/docs/usage/creating-release which is what I think you must have followed when doing the 0.1.0 release of Cake.GitHub:

https://github.com/cake-contrib/Cake.GitHub/releases/tag/0.1.0

**Reply by gep13** [reply]
I was replying quickly on my phone last night, so let my elaborate slightly now that I am at computer...

If you look at the [build](https://ci.appveyor.com/project/cakecontrib/cake-github/builds/28579978) that was done on AppVeyor for the last release, you will see that AppVeyor "knows" that it is a tagged build:

<img width="1480" alt="image" src="https://github.com/user-attachments/assets/f2bc7226-6b33-45f0-b874-92444fd9d5cc">

However, if you look at the [build](https://ci.appveyor.com/project/cakecontrib/cake-github/builds/50392883) that was run last night for the 1.0.0 release, you will see that AppVeyor doesn't know that it was a tag, only that it was run on the `master` branch:

<img width="1490" alt="image" src="https://github.com/user-attachments/assets/63d202aa-fadc-4996-ab16-bff34facdc8f">

The times when I have seen this happen is when I have added a tag via the git command line directly on my machine, and when the tag has been pushed up to GitHub using `git push origin --tags`.  In these situations, the webhook that is sent to AppVeyor from GitHub doesn't include the information about the tag, and as a result Cake.Recipe isn't able to find that information, and as a result, doesn't do the final push of the package(s) to NuGet.

The way that I have found to work, and the way that I think the 0.1.0 release of Cake.GitHub was performed, was by first creating a [release](https://github.com/cake-contrib/Cake.GitHub/releases) on GitHub, and then when you publish that release (which also creates a tag), the webhook that is sent over to AppVeyor, does contain the required information, and Cake.Recipe is able to find this information, and then complete the publishing to NuGet.

I hope this makes sense.  Let me know if you have any questions.

**Reply by GeertvanHorrik** [reply]
That was it. I used my git client (Fork) to handle GitFlow for me. Our own workflow detects the tags from the git repo, but for this I had to manually tag on the GitHub website. It's been a while since I did Cake addin development :smile:

Thanks for all the help, package is now available on NuGet :tada:

**Reply by gep13** [reply]
The Cake.Recipe [build provider for TeamCity](https://github.com/cake-contrib/Cake.Recipe/blob/develop/Source/Cake.Recipe/Content/teamcity.cake) uses information directly from git to determine if a tag exists or not, while the [build provider for AppVeyor](https://github.com/cake-contrib/Cake.Recipe/blob/develop/Source/Cake.Recipe/Content/appveyor.cake) relies on information from AppVeyor itself.

If for no other reason than consistency, perhaps this is something that we should look to change in Cake.Recipe at some point.  Will need to give this some more thought.
